src/gluonts/nursery/streaming_models/rnn/_estimator.py

```diff
@@ -0,0 +1,865 @@
+            t_agg = (pd.Series(t).rolling(self.ratio).agg(self.agg_fun))[
+                self.ratio - 1 :
+            ]
+        else:
+            t = np.concatenate(
+                [buffer["base_target"], data[self.target_field]]
+            )
+            new_agg_lags = (
+                pd.Series(t).rolling(self.ratio).agg(self.agg_fun)
+            )[self.ratio - 1 :]
+            t_agg = pd.Series(
+                np.concatenate([buffer["agg_lags"], new_agg_lags.values])
+            )
+
+        # compute the aggregate lags for each time point of the time series
+        agg_vals = np.concatenate(
+            [
+                np.zeros(
+                    (max(self.valid_lags) * self.ratio + self.half_window,)
+                ),
+                t_agg.values,
+            ],
+            axis=0,
+        )
+        lags = np.vstack(
+            [
+                agg_vals[
+                    -(
+                        l * self.ratio
+                        - self.offset
+                        - self.half_window
+                        + len(data[self.target_field])
+                        - 1
+                    ) : -(l * self.ratio - self.offset - self.half_window - 1)
+                    if -(l * self.ratio - self.offset - self.half_window - 1)
+                    is not 0
+                    else None
+                ]
+                for l in self.valid_lags
+            ]
+        )
+
+        # update the data entry
+        data[self.feature_name] = np.nan_to_num(lags)
+        data[self.lag_state_field] = {
+            "agg_lags": t_agg.values[-self.max_state_lag * self.ratio + 1 :],
+            "base_target": t[-self.ratio + 1 :] if self.ratio > 1 else [],
+        }
+
+        assert data[self.feature_name].shape == (
+            len(self.valid_lags),
+            len(data[self.target_field]),
+        )
+
+        return data
+
+
+class CopyField(SimpleTransformation):
+    """
+    Copies the value of input_field into output_field and does nothing
+    if input_field is not present or None.
+    """
+
+    @validated()
+    def __init__(
+        self,
+        output_field: str,
+        input_field: str,
+    ) -> None:
+        self.output_field = output_field
+        self.input_field = input_field
+
+    def transform(self, data: DataEntry) -> DataEntry:
+
+        field = data.get(self.input_field)
+        if field is not None:
+            data[self.output_field] = data[self.input_field].copy()
+
+        return data
+
+
+class StateInitializer:
+    @validated()
+    def __init__(self, pickled_state: str) -> None:
+        self.pickled_bytes = pickled_state.encode("latin1")
+
+    def __call__(self) -> Any:
+        return pickle.loads(self.pickled_bytes)
+
+    @staticmethod
+    def from_state(state) -> "StateInitializer":
+        s = pickle.dumps(state).decode("latin1")
+        return StateInitializer(s)
+
+
+class StreamingRnnEstimator(Estimator):
```

Comment:
There is the dataset subsampling change that we may or may not want to include now. Just mentioning it here to be aware of what is missing.

Reply:
Should we keep the subsampling code here? It appears to complicate the estimator significantly, and to some extent limit its applicability to `ListDataset` objects (because `len` needs to be invoked)